index.js

```diff
@@ -43,13 +41,30 @@ module.exports = opts => {
     typeof handlers[type] === 'function' ? handlers[type] : missing(type)
 
   const processJob = ({ type, payload }, callback) => {
-
-    const done          = once(callback)
-          details       = mergeAll([ options, { type, payload }, timeoutErr ]),
-          logTimeout    = partial(timeoutLogger, [ details ]),
-          handleTimeout = compose(partial(done, [ new Error(timeoutErr.error) ]), tap(logTimeout))
-          timeout       = setTimeout(handleTimeout, visibilityTimeout * 1000),
-          finish        = compose(tap(partial(clearTimeout, [ timeout ])), done)
+    const done = once(callback)
+
+    const addErrorContext = error => {
+      error.params = mergeAll([ options, { type, payload }, timeoutErr ])
+      error.component = 'squiss-jobs'
+      return error
+    }
+
+    const complete = error => {
+      const doneError = compose(done, when(is(Error), addErrorContext))
+      return arguments.length === 0
+        ? done()
+        : doneError(error)
+    }
+
+    const handleTimeout = () => {
+      const error = new Error(timeoutErr.error)
+      addErrorContext(error)
+      timeoutLogger(params)
```

Comment:
I don't think there's a `params` in scope at this point...which is probably my fault for suggesting you combine those two lines earlier where iirc, you had a separate `const params = ...`.

Reply:
Yeah, I'm not seeing any `params` anywhere.  Do we have linting for this library?